### ISSUE TITLE
PRS error when dropping index in msyql

### ISSUE THREAD

**Issue by ngdah** [sqlfluff/sqlfluff#1723]
## Expected Behaviour
SQLFluff reporting issues on `DROP INDEX` statement

## Observed Behaviour
SQLFluff reports `PRS: Found unparsable section`

## Steps to Reproduce

- Create file `drop_index.sql`
```sql
DROP INDEX `test` ON `table`.`foo`;

```

- Run SQLFluff lint command
```
sqlfluff lint --dialect=mysql drop_index.sql
```
Received output:
```
== [drop_index.sql] FAIL
L:   1 | P:  18 |  PRS | Line 1, Position 18: Found unparsable section: ' ON
                       | `table`.`foo`'
All Finished 📜 🎉!
```

## Dialect
MySQL

## Version
Python 3.9.7
sqlfluff, version 0.6.9

## Configuration
No additional config provided, all default settings

**Comment by fatelei** [comment]
currently，the syntax of drop index is `DROP INDEX <Index name> [CONCURRENTLY] [IF EXISTS] {RESTRICT | CASCADE}`

mysql dialect should replace the drop index segment, i will submit a pr
